.github/workflows/ci.yaml

```diff
@@ -217,11 +217,43 @@ jobs:
           name: buildtools
           path: output/bin/buildtools
 
+  should-run-e2e:
+    name: Should run e2e
+    runs-on: ubuntu-latest
+    outputs:
+      run-e2e: ${{ steps.changed-files.outputs.e2e_any_changed }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v46.0.5
```

Comment:
There was a security incident earlier this year with a `changed-files` action where every release tag got updated to point to a malicious commit somehow. I think in all our repos where we were using it, we updated to point to a commit hash instead of a semver tag:

https://github.com/replicatedcom/gitops-deploy/pull/1836
https://www.stepsecurity.io/blog/harden-runner-detection-tj-actions-changed-files-action-is-compromised

This is more a general gh action security best practice thing and obviously creates more tech debt on our end, just thought i'd bring it up since the action name looked familiar.